crypto/packet.c

```diff
@@ -65,15 +65,21 @@ int WPACKET_reserve_bytes(WPACKET *pkt, size_t len, unsigned char **allocbytes)
         if (BUF_MEM_grow(pkt->buf, newlen) == 0)
             return 0;
     }
-    if (allocbytes != NULL)
+    if (allocbytes != NULL) {
         *allocbytes = WPACKET_get_curr(pkt);
+        if (pkt->endfirst)
```

Comment:
Should be `if (pkt->endfirst && *allocbytes != NULL)`

Reply:
Yes - I actually had this fix in my other temporary branch that I shared with you - but it hadn't made it back into this PR